src/components/AnnualCycleGraph/AnnualCycleGraph.js

```diff
@@ -9,33 +9,49 @@ const Plot = createPlotlyComponent(Plotly);
 
 
 function AnnualCycleGraph({annualData}) {
-    const monthAbbrev = [
-        "Jan", "Feb", "Mar", "Apr", 
-        "May", "Jun", "Jul", "Aug", 
-        "Sep", "Oct", "Nov", "Dec"];
     
+    var yAxisTitle = `Mean Maximum Temperature (${ annualData.units })`;
+
+    const months = keys(annualData.data).sort();
+        
     function monthlyTimeSeries() {
         if(annualData === null){
             return []
         }
         else {
             //TODO: fix this, order is not guarenteed
-            return values(annualData.data);
+            return matchValues(entries(annualData.data), months);
         }
     }
 
     return (
         <Plot
             data={[
                 {
-                    x: monthAbbrev,
+                    x: months,
                     y: monthlyTimeSeries(),
                     type: 'scatter',
                     mode: 'lines+markers',
                     marker: {color: 'red'},
+                    hovertemplate: '%{y:.2f}\u00B0C<extra></extra>',
```

Comment:
This looks great, but could you use the units from the API call instead of hardcoding it to degrees?